noir-projects/aztec-nr/aztec/src/history/nullifier_inclusion.nr

```diff
@@ -3,62 +3,31 @@ use dep::protocol_types::header::Header;
 
 use crate::{
     context::{PrivateContext, ContextInterface},
-    oracle::get_nullifier_membership_witness::get_nullifier_membership_witness,
+    oracle::get_nullifier_membership_witness::{get_nullifier_membership_witness, NullifierMembershipWitness},
     note::{utils::compute_siloed_nullifier, note_interface::NoteInterface}
 };
 
-fn _nullifier_inclusion(nullifier: Field, header: Header) {
+/**
+ * This function checks if a nullifier is included in the nullifier tree at a given header.
+ * IMPORTANT: If the function returns false, you don't have a proof of non-inclusion.
+ *
+ * @param nullifier The nullifier to prove inclusion of
+ * @param header The header at which we'll prove that the nullifier exists in the nullifier tree
+ * @return true if the nullifier is included in the nullifier tree, false otherwise
+ */
+pub fn nullifier_valid_inclusion(nullifier: Field, header: Header) -> bool {
```

Comment:
I'm not really following why we are changing these libraries? Was it not just the context you wanted to have the `exists` that matches?